dot/network/block_announce.go

```diff
@@ -183,6 +188,10 @@ func (s *Service) validateBlockAnnounceHandshake(from peer.ID, hs Handshake) err
 		return errors.New("invalid handshake type")
 	}
 
+	if int(bhs.Roles) > 4 {
+		return errors.New("invalid handshake role")
```

Comment:
Use a sentinel error, and wrap the actual role with it

```suggestion
		return fmt.Errorf("%w: %d", errHandshakeRoleNotValid, bhs.Roles)
```

Reply:
changed to enum now